### DISCUSSION TITLE
How to update pre-release branch about latest version that released in main/master branch?

### DISCUSSION THREAD

**Discussion by isabr85** [semantic-release/semantic-release#2268] [Configuration]
I have 2 branches - dev for pre-release version and master for release.

let’s say that both are beginning with version 1.0.0

when I merge my commits into dev, semantic release generate version of 1.0.1-rc.1

then I am merging dev into master and it creates version of 1.0.1

at this point - dev latest tag version is 1.0.1-rc.1 so at the next merge to dev it will generates a pre-release version of 1.0.1-rc.2 instead of 1.0.2-rc.1 because 1.0.1 version tag is inside master branch

How do I update dev branch about the released version after master generates a new version as part of semantic release solution without doing manually tags merging between the branches?

thanks

**Comment by gr2m** [comment]
the `dev` branch is not aware of the `1.0.1` tag, it only exists in `master`. Usually I'd delete a branch for pre-releases after I merged it into the main branch, then re-create it as needed. 

`semantic-release` won't work with a long living `dev` branch for pre releases.
